### PR TITLE
Fix for when 'trailer' is indented

### LINKED ISSUE #214: multiple xrefs in pdf cause page extraction return only the first page
PDF file: [bad_page_number.pdf](https://github.com/pdfminer/pdfminer.six/files/2672072/bad_page_number.pdf)
- pdfminer extracts only the first page of the file 
- parser.bufpos stop at size 10325
- it looks like this file contains two `%%EOF` lines

```python
from pdfminer.pdfparser import PDFParser
from pdfminer.pdfdocument import PDFDocument
from pdfminer.pdfpage import PDFPage
from pdfminer.pdfinterp import PDFResourceManager
from pdfminer.pdfinterp import PDFPageInterpreter
from pdfminer.pdfdevice import PDFDevice
from pdfminer.layout import LAParams
from pdfminer.converter import PDFPageAggregator
from pdf2html import *
fpath = '~/bad_page_number.pdf.pdf'
fp = open(fpath, 'rb')
parser = PDFParser(fp)
laparams = LAParams()
document = PDFDocument(parser)
rsrcmgr = PDFResourceManager()
device = PDFPageAggregator(rsrcmgr, laparams=laparams)
interpreter = PDFPageInterpreter(rsrcmgr, device)
print(len(list(PDFPage.create_pages(document))))
```

Could you please look into the problem please?

### PR DESCRIPTION
**Pull request by jstockwin**
**Pull request**

Thanks for improving pdfminer.six! Please include the following information to
help us discuss and merge this PR:

Fix a bug where `trailer` being indented caused a bug. Thanks to @pietermarsman for suggesting the fix.

Closes #214

**How Has This Been Tested?**

There is an example script and PDF in #214.

**Checklist**

- [x] I have added tests that prove my fix is effective or that my feature 
  works  (N/A -> we don't really want to add this PDF to our test suite, correct?)
- [X] I have added docstrings to newly created methods and classes  (N/A)
- [X] I have optimized the code at least one time after creating the initial 
  version
- [X] I have updated the [README.md](../README.md) or I am verified that this
  is not necessary
- [X] I have updated the [readthedocs](../docs/source) documentation or I 
  verified that this is not necessary
- [X] I have added a consice human-readable description of the change to 
  [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

File: pdfminer/pdfdocument.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def load(self, parser):
                 raise PDFNoValidXRef('Unexpected EOF - file corrupted?')
             if not line:
```
Comment (pietermarsman) [inline pdfminer/pdfdocument.py:100]: Can you change this check to `len(line) == 0`, or `line is None or len(line) == 0`

File: pdfminer/pdfdocument.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def load(self, parser):
...
                 raise PDFNoValidXRef('Premature eof: %r' % parser)
-            if line.startswith(b'trailer'):
+            if line.strip().startswith(b'trailer'):
```
Comment (pietermarsman) [inline pdfminer/pdfdocument.py:102]: Everytime `line` is used it is stripped, so I think its best to strip the whitespace right after parsing on line 95.

**Review by pietermarsman — CHANGES_REQUESTED** [review]

File: pdfminer/pdfdocument.py (hunk 1 above)
Comment (jstockwin) [reply pdfminer/pdfdocument.py:100]: Actually @pietermarsman I struggle to see why this line is needed at all.

On line 96 we already do
```
                if not line.strip():
                    continue
```
so I'm pretty sure this condition will never triggered. Do you agree?

Comment (pietermarsman) [reply pdfminer/pdfdocument.py:100]: Yes

Comment (jstockwin) [reply pdfminer/pdfdocument.py:100]: Cool, I'll remove it and also address your other comment today :+1:

**Comment by jstockwin** [comment]
@pietermarsman I just pushed a new commit addressing your code review :+1:

**Comment by pietermarsman** [comment]
Oops! Merged this while it was broken...

**Pull request merged**